hyperspy/_signals/eds_tem.py

```diff
@@ -343,9 +371,25 @@ def quantification(self,
                 print("%s (%s): Composition = %.2f %s percent"
                       % (element, xray_line, composition[i].data,
                          composition_units))
+        if method=='cross_section':
+            for i, xray_line in enumerate(xray_lines):
+                element, line = utils_eds._get_element_and_line(xray_line)
+                number_of_atoms[i].metadata.General.title = 'atom counts of ' +\
+                    element
+                number_of_atoms[i].metadata.set_item("Sample.elements",
+                    ([element]))
+                number_of_atoms[i].metadata.set_item(
+                    "Sample.xray_lines", ([xray_line]))
         if plot_result and composition[i].axes_manager.signal_dimension != 0:
             utils.plot.plot_signals(composition, **kwargs)
-        return composition
+        if method=='zeta':
+            self.metadata.set_item("Sample.mass_thickness", mass_thickness)
+        if method == 'zeta':
+            return composition, mass_thickness
```

Comment:
If both of the above are done if method == 'zeta' then you don't need two if statements.